src/components/Karma/utils.ts

```diff
@@ -0,0 +1,21 @@
+export const renderCircleColor = (karma: number) => {
+  if (karma > 87.5) {
+    return ['#50BA57', '#C7EECA', '#50BA57'];
+  }
+  if (karma > 75) {
+    return ['#70BF21', '#B3E57F', '#70BF21'];
+  }
+  if (karma > 62.5) {
+    return ['#A0C527', '#D1EA82', '#A0C527'];
+  }
+  if (karma > 50) {
+    return ['#D3D20C', '#F3F370', '#D3D20C'];
+  }
+  if (karma > 37.5) {
+    return ['#E4B92E', '#FFE591', '#E4B92E'];
+  }
+  if (karma > 25) {
+    return ['#F4833C', '#F9BE9A', '#F4833C'];
+  }
+  return ['#DF001F', '#F5B0B9', '#DF001F'];
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `renderCircleColor` function could be optimized by using a data structure like an array of objects or tuples, where each object or tuple represents a range and its corresponding color. This would make the code more maintainable and easier to modify in the future. Here's how you could refactor it:

```diff
- export const renderCircleColor = (karma: number) => {
-   if (karma > 87.5) {
-     return ['#50BA57', '#C7EECA', '#50BA57'];
-   }
-   if (karma > 75) {
-     return ['#70BF21', '#B3E57F', '#70BF21'];
-   }
-   if (karma > 62.5) {
-     return ['#A0C527', '#D1EA82', '#A0C527'];
-   }
-   if (karma > 50) {
-     return ['#D3D20C', '#F3F370', '#D3D20C'];
-   }
-   if (karma > 37.5) {
-     return ['#E4B92E', '#FFE591', '#E4B92E'];
-   }
-   if (karma > 25) {
-     return ['#F4833C', '#F9BE9A', '#F4833C'];
-   }
-   return ['#DF001F', '#F5B0B9', '#DF001F'];
- };

+ const karmaColors = [
+   { limit: 87.5, colors: ['#50BA57', '#C7EECA', '#50BA57'] },
+   { limit: 75, colors: ['#70BF21', '#B3E57F', '#70BF21'] },
+   { limit: 62.5, colors: ['#A0C527', '#D1EA82', '#A0C527'] },
+   { limit: 50, colors: ['#D3D20C', '#F3F370', '#D3D20C'] },
+   { limit: 37.5, colors: ['#E4B92E', '#FFE591', '#E4B92E'] },
+   { limit: 25, colors: ['#F4833C', '#F9BE9A', '#F4833C'] },
+   { limit: 0, colors: ['#DF001F', '#F5B0B9', '#DF001F'] }
+ ];
+
+ export const renderCircleColor = (karma: number) => {
+   for (let i = 0; i < karmaColors.length; i++) {
+     if (karma > karmaColors[i].limit) {
+       return karmaColors[i].colors;
+     }
+   }
+ };
```
This way, you can easily add, remove, or modify ranges and their corresponding colors by just modifying the `karmaColors` array.


<!-- This is an auto-generated comment by OSS CodeRabbit -->